src/core/qfieldcloudprojectsmodel.cpp

```diff
@@ -359,262 +403,596 @@ void QFieldCloudProjectsModel::cancelDownloadProject( const QString &projectId )
+    }
+    else
+    {
+      project->errorStatus = NoErrorStatus;
+      project->packagingStatus = PackagingFinishedStatus;
+      project->packagingStatusString = QString();
+    }
 
-            if ( !layer.value( QStringLiteral( "valid" ) ).toBool() )
-            {
-              QString layerName = layer.value( QStringLiteral( "name" ) ).toString();
-              QString layerStatus = layer.value( QStringLiteral( "status" ) ).toString();
+    project->status = ProjectStatus::Idle;
 
-              project->packagedLayerErrors.append( tr( "Packaged layer '%1' is not valid: '%2'" )
-                                                     .arg( layerName )
-                                                     .arg( layerStatus ) );
-              QgsMessageLog::logMessage( project->packagedLayerErrors.last() );
+    emit dataChanged( projectIndex, projectIndex, QVector<int>() << StatusRole << PackagingStatusRole << ErrorStatusRole << ErrorStringRole );
+    emit projectDownloaded( projectId, project->name, hasError, errorString );
+  } );
+}
 
-              hasLayerExportErrror = true;
-            }
-          }
+void QFieldCloudProjectsModel::projectDownload( const QString &projectId )
+{
+  QgsLogger::debug( QStringLiteral( "Project %1: downloading..." ).arg( projectId ) );
 
-          if ( hasLayerExportErrror )
-          {
-            QgsLogger::debug( QStringLiteral( "Packaged files list request finished for \"%1\" with some failed layers:\n%2" ).arg( projectId, project->packagedLayerErrors.join( QStringLiteral( "\n" ) ) ) );
-            emit dataChanged( projectIndex, projectIndex, QVector<int>() << PackagedLayerErrorsRole );
-          }
+  if ( !mCloudConnection )
+    return;
 
-          QgsLogger::debug( QStringLiteral( "Packaged files to download - %1 files, namely: %2" )
-                              .arg( project->downloadFileTransfers.count() )
-                              .arg( project->downloadFileTransfers.keys().join( ", " ) ) );
+  const QModelIndex projectIndex = findProjectIndex( projectId );
+  if ( !projectIndex.isValid() )
+    return;
 
-          updateActiveProjectFilesToDownload( projectId );
-          projectDownloadFiles( projectId );
-        } );
-        break;
+  CloudProject *project = mProjects[projectIndex.row()];
+
+  if ( project->isPackagingAborted )
+    return;
+
+  NetworkReply *reply = mCloudConnection->get( QStringLiteral( "/api/v1/packages/%1/latest/" ).arg( projectId ) );
+
+  emit dataChanged( projectIndex, projectIndex, QVector<int>() << PackagingStatusRole << StatusRole );
+
+  connect( reply, &NetworkReply::finished, reply, [=]() {
+    if ( !findProject( projectId ) )
+    {
+      QgsLogger::debug( QStringLiteral( "Project %1: adding download file connections, but the project is deleted." ).arg( projectId ) );
+      return;
+    }
+
+    if ( project->isPackagingAborted )
+      return;
```

Comment:
You have  been throwing a logger debug message on other aborted scenarios, do you want one there? Also, emit something? (I'm starting to feel there's no need for such an emit in this abort scenario anywhere :) )

Reply:
Yes, aborts should be handled by the caller. Added comments.